ui/components/YamlView.tsx

```diff
@@ -1,25 +1,21 @@
 import * as React from "react";
-import styled from "styled-components";
 import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
 import { darcula } from "react-syntax-highlighter/dist/cjs/styles/prism";
+import styled from "styled-components";
+import { AppContext, ThemeTypes } from "../contexts/AppContext";
 import { ObjectRef } from "../lib/api/core/types.pb";
 import { createYamlCommand } from "../lib/utils";
 import CopyToClipboard from "./CopyToCliboard";
 
-export enum UiMode {
-  Light = "Light",
-  Dark = "Dark",
-}
-
 export type YamlViewProps = {
   className?: string;
   yaml: string;
   object?: ObjectRef;
-  mode?: UiMode;
+  theme?: ThemeTypes;
```

Comment:
There's currently some work going on [here](https://github.com/weaveworks/weave-gitops-enterprise/pull/2826) where we wanted the ability to override the theme in just the YamlView. The story is developing so this may not be necessary soon.